Source/engine/render/text_render.cpp

```diff
@@ -420,9 +427,57 @@ uint32_t DoDrawString(const Surface &out, std::string_view text, Rectangle rect,
 		}
 	};
 
-	for (; !remaining.empty() && remaining[0] != '\0'
-	     && (next = DecodeFirstUtf8CodePoint(remaining, &cpLen)) != Utf8DecodeError;
-	     remaining.remove_prefix(cpLen)) {
+	const auto drawSingleLine = [&]() {

```

Comment:
It's a little bit unclear at a glance what this lambda captures, consider extracting the contents of the lambda into a function.